spring-cql/src/test/java/org/springframework/cassandra/test/integration/support/CassandraConnectionProperties.java

```diff
@@ -80,17 +79,47 @@ public long getCqlInitializationTimeout() {
 		return getLong("build.cql.init.timeout");
 	}
 
+	public String getCassandraHost() {
+		return getProperty("build.cassandra.host");
+	}
+
+	public CassandraType getCassandraType() {
+		String property = getProperty("build.cassandra.mode");
+		if (property != null && property.equalsIgnoreCase(CassandraType.EXTERNAL.name())) {
+			return CassandraType.EXTERNAL;
+		}
+		return CassandraType.EMBEDDED;
+	}
+
 	public int getInt(String key) {
 		String property = getProperty(key);
-		return Integer.parseInt(property);
+		try {
+			return Integer.parseInt(property);
+		} catch (NumberFormatException e) {
+			throw new IllegalStateException(resourceName + ": Cannot parse property " + key + " to int: " + property);
+		}
 	}
 
 	public long getLong(String key) {
 		String property = getProperty(key);
-		return Long.parseLong(property);
+		try {
+			return Long.parseLong(property);
+		} catch (NumberFormatException e) {
+			throw new IllegalStateException(resourceName + ": Cannot parse property " + key + " to long: " + property + "");
+		}
 	}
 
 	public boolean getBoolean(String key) {
-		return Boolean.parseBoolean(getProperty(key));
+		String property = getProperty(key);
+		try {
+			return Boolean.parseBoolean(property);
+		} catch (NumberFormatException e) {
```

Comment:
Does not throw a `NumberFormatException` here; just returns a true or false depending whether the String value representing a boolean is actually equal to "true", case insensitive.